python/hail/representation/genomeref.py

```diff
@@ -185,6 +185,46 @@ def GRCh38(cls):
         """
         return GenomeReference._from_java(Env.hail().variant.GenomeReference.GRCh38())
 
+    @classmethod
+    @record_classmethod
+    @handle_py4j
+    @typecheck_method(file=strlike)
+    def from_file(cls, file):
+        """Load reference genome from a JSON file.
+
+        The JSON file must have the following format:
+
+        .. code-block:: text
+
+            {"name": "my_reference_genome",
+             "source": "/my/source",
+             "contigs": [{"name": "1", "length": 10000000},
+                         {"name": "2", "length": 20000000},
+                         {"name": "X", "length": 19856300},
+                         {"name": "Y", "length": 78140000},
+                         {"name": "MT", "length": 532}],
+             "xContigs": ["X"],
+             "yContigs": ["Y"],
+             "mtContigs": ["MT"],
+             "par": [{"start": {"contig": "X","position": 60001},"end": {"contig": "X","position": 2699521}},
+                     {"start": {"contig": "Y","position": 10001},"end": {"contig": "Y","position": 2649521}}]
+            }
+
+        **Notes**
+
+        `name` must be unique and not overlap with Hail's pre-instantiated references: "GRCh37" and "GRCh38". `source` is
+        intended to be a string representing the source of the JSON file contents.
```

Comment:
What is `source` for?

Reply:
I deleted it. It's not required. I had it in the json files that are in the hail directory to remember where I got the data from.